bigtable-hbase/src/main/java/com/google/cloud/bigtable/hbase/BigtableBufferedMutator.java

```diff
@@ -91,48 +83,22 @@ public BigtableBufferedMutator(
       String dataHost,
       BufferedMutator.ExceptionListener listener,
       ExecutorService heapSizeExecutor) {
-    this.client = client;
     this.adapter = adapter;
-    this.sizeManager = new HeapSizeManager(maxHeapSize, maxInflightRpcs, heapSizeExecutor);
     this.configuration = configuration;
     this.exceptionListener = listener;
     this.host = dataHost;
+    this.asyncMutator = new AsyncMutator(client, maxInflightRpcs, maxHeapSize, heapSizeExecutor);
   }
 
   @Override
   public void close() throws IOException {
-    WriteLock lock = mutationLock.writeLock();
-    lock.lock();
-    try {
-      if (!closed) {
-        doFlush();
-        closed = true;
-      }
-    } finally {
-      lock.unlock();
-    }
+    flush();
```

Comment:
Angus and I went back and forth on this one.  We should always flush, even if a user previously closed; the user may have had an exception that happened in a previous call to close, so try to complete the calls again.